### PR TITLE
Support rolling-update of a single instance group

### LINKED ISSUE #1265: rolling-update should support a single instancegroup
Rather than requiring a full cluster update

### PR DESCRIPTION
**Pull request by justinsb**
Fix #1265

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1270)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
How badly will this smash my work on rolling updates?

**Comment by justinsb** [comment]
I would hope not at all - we only rolling update on some IGs

**Comment by justinsb** [comment]
Rebased

File: cmd/kops/rollingupdatecluster.go [hunk 1]
```diff
@@ -32,7 +34,7 @@ import (
 	"time"
 )
 
-type RollingUpdateClusterCmd struct {
+type RollingUpdateOptions struct {
```
Comment (kris-nova) [inline cmd/kops/rollingupdatecluster.go]: We really need to do a sweep of our cobra commands..

File: cmd/kops/rollingupdatecluster.go [hunk 2]
```diff
@@ -41,49 +43,75 @@ type RollingUpdateClusterCmd struct {
...
+}
+
+func (o *RollingUpdateOptions) InitDefaults() {
+	o.Yes = false
+	o.Force = false
+	o.CloudOnly = false
+
+	o.MasterInterval = 5 * time.Minute
+	o.NodeInterval = 2 * time.Minute
+	o.BastionInterval = 5 * time.Minute
```
Comment (kris-nova) [inline cmd/kops/rollingupdatecluster.go]: We can probably turn this down - bastions literally do nothing but run an SSH daemon

**Comment by erictune** [comment]
Does a kops rolling update do drains on nodes, using /evict?  Couldn't find any mention of this in the docs.

**Comment by erictune** [comment]
Sorry, thats not really the point of this PR.  I'll ask on the mailing list.

**Pull request merged**